validation/src/lib.rs

```diff
@@ -151,14 +152,28 @@ pub fn sign_table_statement(statement: &Statement, key: &ValidatorPair, parent_h
 
 /// Check signature on table statement.
 pub fn check_statement(statement: &Statement, signature: &ValidatorSignature, signer: ValidatorId, parent_hash: &Hash) -> bool {
-	use runtime_primitives::traits::AppVerify;
-
 	let mut encoded = PrimitiveStatement::from(statement.clone()).encode();
 	encoded.extend(parent_hash.as_ref());
 
 	signature.verify(&encoded[..], &signer)
 }
 
+/// Sign an erasure chunk message.
+pub fn sign_chunk(chunk: &ErasureChunk, key: &ValidatorPair, parent_hash: &Hash) -> ValidatorSignature {
+	let mut encoded = chunk.encode();
+	encoded.extend(parent_hash.as_ref());
+
+	key.sign(&encoded)
+}
+
+/// Check the signature of a chunk message.
+pub fn check_chunk(chunk: &ErasureChunk, signature: &ValidatorSignature, signer: ValidatorId, parent_hash: &Hash) -> bool {
```

Comment:
the signature should be on the raw chunk data, if anything. But I don't really see the need for a signature in the gossip protocol. This and `sign_chunk` should probably just be removed.